### PR TITLE
feat(kubernetes): switch to monthly billing

### PR DESCRIPTION
**Pull request by radireddy**
show node billing type and provide an option to switch node billing type to monthly

Closes #MBP-354

### CODE REVIEW

File: packages/manager/modules/kubernetes/src/details/nodes/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -43,6 +43,17 @@
   "kube_nodes_common_add": "Ajouter",
   "kube_nodes_common_cancel": "Annuler",
   "kube_nodes_common_delete": "Supprimer",
+  "kube_nodes_common_confirm": "Confirmer",
   "kube_nodes_common_field_error_required": "Veuillez compléter ce champ.",
-  "kube_service_cluster_status": "Statut"
+  "kube_service_cluster_status": "Statut",
+  "kube_nodes_instances_error": "Une erreur s'est produite lors du chargement des détails de facturation: {{ message }}",
```
Comment (antleblanc) [inline packages/manager/modules/kubernetes/src/details/nodes/translations/Messages_fr_FR.json]:
```suggestion
  "kube_nodes_instances_error": "Une erreur s'est produite lors du chargement des détails de facturation : {{ message }}",
```

File: packages/manager/modules/kubernetes/src/details/nodes/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -43,6 +43,17 @@
...
+  "kube_nodes_billing_type": "Type de facturation",
+  "kube_nodes_billing_monthly": "Mensuel",
+  "kube_nodes_billing_hourly": "Toutes les heures",
+  "kube_nodes_billing_monthly_pending": "Passage au mensuel",
+  "kube_nodes_switch_to_monthly_billing_title": "Passer à la facturation mensuelle",
+  "kube_nodes_switch_to_monthly_billing_desc": "Êtes-vous sûr de vouloir basculer la facturation du nœud {{instanceName}} vers la facturation mensuelle?",
```
Comment (antleblanc) [inline packages/manager/modules/kubernetes/src/details/nodes/translations/Messages_fr_FR.json]:
```suggestion
  "kube_nodes_switch_to_monthly_billing_desc": "Êtes-vous sûr de vouloir basculer la facturation du nœud {{instanceName}} vers la facturation mensuelle ?",
```

File: packages/manager/modules/kubernetes/src/details/nodes/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -43,6 +43,17 @@
...
+  "kube_nodes_switch_to_monthly_billing_info": "Vous serez immédiatement facturé au prorata des jours restants du mois en cours.",
+  "kube_nodes_switch_billing_type_success": "Le type de facturation du nœud est modifié en facturation mensuelle",
+  "kube_nodes_switch_billing_type_error": "Impossible de passer à la facturation mensuelle: {{message}}"
```
Comment (antleblanc) [inline packages/manager/modules/kubernetes/src/details/nodes/translations/Messages_fr_FR.json]:
```suggestion
  "kube_nodes_switch_billing_type_error": "Impossible de passer à la facturation mensuelle : {{message}}"
```

**Comment by antleblanc** [comment]
@radireddy Pay attention that the module has been moved.

`packages/manager/modules/kubernetes/src` → `packages/manager/modules/pci/src/projects/project/kubernetes`

I think you should rebase into `public-cloud` branch instead and prefer open Pull request directly into it.

ref: https://github.com/ovh-ux/manager/pull/371

File: packages/manager/modules/kubernetes/src/details/nodes/billing-type/template.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<form id="swithToMonthlyBilling"
```
Comment (FredericEspiau) [inline packages/manager/modules/kubernetes/src/details/nodes/billing-type/template.html]:
```suggestion
<form
```
No need for this

File: packages/manager/modules/kubernetes/src/details/nodes/billing-type/template.html [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+      name="swithToMonthlyBilling"
```
Comment (FredericEspiau) [inline packages/manager/modules/kubernetes/src/details/nodes/billing-type/template.html]:
```suggestion
      name="$ctrl.swithToMonthlyBilling"
```

File: packages/manager/modules/kubernetes/src/details/nodes/controller.js [hunk 6]
```diff
@@ -68,6 +71,29 @@ export default class KubernetesNodesCtrl {
...
+        const monthlyBilling = get(instance, 'monthlyBilling');
+        if (isEmpty(monthlyBilling)) {
+          set(node, 'billingType', 'hourly');
+        } else if (monthlyBilling.status === 'ok') {
+          set(node, 'billingType', 'monthly');
+        } else {
+          set(node, 'billingType', 'monthly_pending');
+        }
+      })
+      .catch(() => this.CucCloudMessage.error(this.$translate.instant('kube_nodes_instances_error')));
```
Comment (FredericEspiau) [inline packages/manager/modules/kubernetes/src/details/nodes/controller.js]:
```suggestion
      .catch(() => {
        this.CucCloudMessage.error(this.$translate.instant('kube_nodes_instances_error')));
      }
```

**Pull request closed without merging**